pkg/action/executor/gitops/git_manager.go

```diff
@@ -97,12 +97,10 @@ func (r *GitManager) Update(replicas int64, podSpec map[string]interface{}) (Wai
 	}
 
 	switch {
-	case strings.HasPrefix(url.Host, "github."):
-		if url.Host == "github.com" {
-			handler = NewGitHubHandler(gitHandler, getClient(ctx, token, ""))
-		} else {
-			handler = NewGitHubHandler(gitHandler, getClient(ctx, token, url.Scheme+"://"+url.Host))
-		}
+	case url.Host == "github.com": // Public GitHub
+		handler = NewGitHubHandler(gitHandler, getClient(ctx, token, ""))
+	case strings.HasPrefix(url.Host, "github."): // Enterprise GitHub
```

Comment:
The issue with this is that public github `github.com` also match this case

Reply:
AFAIK Ideally golang switch case evaluation is that the first one that equals the switch expression triggers execution of the statements of the associated case and the rest are ignored, so this might be ok, but this might need some testing to make sure that this indeed is the behaviour.

Reply:
Wrote a simple test to verify that it will only match on the first case that matches.

Reply:
I prefer the first commit